src/app/server/Dnssd.cpp

```diff
@@ -489,5 +489,12 @@ CHIP_ERROR DnssdServer::GenerateRotatingDeviceId(char rotatingDeviceIdHexBuffer[
 }
 #endif
 
+void DnssdServer::OnICDModeChange()
+{
+    // ICDMode changed, restart DNS-SD advertising, because SII and ICD key are affected by this change.
+    // StartServer will take care of setting the operational and commissionable advertissements
+    StartServer();
```

Comment:
Do you know if start servers created a new entry or updates the existing one?

Reply:
Remove old and and create a new one with same instance or hostname so it guess updated in the DNS adv directly